src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,132 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
-
+        if (size == 0) {
+            head = new Node<>(null, value, null);
+            tail = head;
+            size++;
+            return;
+        }
+        tail = new Node<>(tail, value, null);
+        tail.prev.next = tail;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == index) {
+            add(value);
+            return;
+        }
+        Node<T> clone = returnNodeByIndex(index);
+        Node<T> newNode = new Node<>(clone.prev, value, clone);
+        clone.prev.next = newNode;
+        clone.prev = newNode;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (T i : list) {
+            add(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return returnNodeByIndex(index).value;
     }
 
     @Override
     public void set(T value, int index) {
-
+        returnNodeByIndex(index).value = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException(index);
+        }
+        if (index == 0) {
+            T value = head.value;
+            head = head.next;
+            size--;
+            return value;
+        }
+        if (index == size - 1) {
+            final T value = tail.value;
+            tail = tail.prev;
+            tail.next = null;
+            size--;
+            return value;
+        }
+        Node<T> willBeRemove = returnNodeByIndex(index);
+        willBeRemove.prev.next = willBeRemove.next;
+        willBeRemove.next.prev = willBeRemove.prev;
+        size--;
+        return willBeRemove.value;
     }
 
     @Override
     public T remove(T t) {
```

Comment:
where do you remove the value?

Reply:
I do not understand the question. Need to reset the link to this object?

Reply:
Added 
 willBeRemove.next = null;
 willBeRemove.prev = null;

Reply:
Is that what you meant?